### PR TITLE
Feature#4 crud for emloyee

### LINKED ISSUE #4: Manage employees (CRUD)
# REQUIREMENTS

**Manage employees (CRUD operations)**

- You can see the employees from each company branch.
- When creating or updating an employee, the name, email and position are assignable and updatable.
- When creating an employee company branch, employee number and private number are assigned automatically.
- The views where made for each operation of CRUD (except Delete) and the navigation is through buttons.
- Updated company branch views for the styles to be the same.

# DESIGN

bootstraps example:
https://getbootstrap.com/docs/5.0/content/tables/

Image 2021-04-12 at 5 00 44 p m

# Acceptance Criteria
- [ ] Admin can create employee.
- [ ] Show list employees in a table.
- [ ] Admin can show one employee.
- [ ] Can edit employee.

### PR DESCRIPTION
**Pull request by RobertVel666**
# Description

- You can see the employees from each company branch.
- When creating or updating an employee, the name, email and position are assignable and updatable.
- When creating an employee company branch, employee number and private number are assigned automatically.
- The views were made for each operation of CRUD (except Delete) and the navigation is through buttons.
- Updated company branch views for the styles to be the same. 

 # Testing

After cloning the repository run on the terminal in the repository directory:
- `yarn install`
- `bundle install`
- `rails db:setup`
- `rails db:seed`
- `rails server`

- Enter the URL (e.g.: http://127.0.0.1:3000) that rails server is listening on in your navigator.
- User: admin@example.com
- Password: password

- Press the  **show** button of any company branch.

Now you can manage your employees.

 ## Screenshots
**Walkthrough**
employee_test_1

### CODE REVIEW

**Review by Arkamis — COMMENTED** [review]
Looks great, you have like 2 views with cols without row. Shouldn't you wrap all your cols inside a row tag?. And what's the difference between private number and employee number?

**Comment by RobertVel666** [comment]
> Looks great, you have like 2 views with cols without row. Shouldn't you wrap all your cols inside a row tag?. And what's the difference between private number and employee number?

Yeah there are still some styles to resolve but we will be addressing them in later issues, and the private number is a number only the employee knows and is for registering the check-in/out, the employee number is just a random number assigned to an employee.

File: app/views/company_branches/new.html.erb [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
-  <div class="mb-3">
-    <%= form.label :address%>
-    <%= form.text_field :address, class: "form-control"%>
-  </div>
-  <div class="mb-3">
-    <%= form.submit "Add company", class: "btn btn-primary" %>
-    <%= link_to "Back", company_branches_path , method: :get, class: "btn btn-primary" %>
-  </div>
-<% end %>
+<div class="col-5">
```
Comment (Arkamis) [inline app/views/company_branches/new.html.erb:1]: You should wrap this inside a container and a row.
See:  [bootstrap docs](https://getbootstrap.com/docs/5.0/layout/grid/).

File: app/views/employees/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="col-lg-5">
```
Comment (Arkamis) [inline app/views/employees/new.html.erb:1]: Same here

File: db/migrate/20210415203352_create_employees.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ def change
       t.string :name
       t.string :email
       t.string :position
+      t.integer :employee_number
```
Comment (Arkamis) [inline db/migrate/20210415203352_create_employees.rb:7]: May I ask, What's the difference between this field and the private number field?

File: app/views/company_branches/new.html.erb (hunk 1 above)
Comment (RobertVel666) [reply app/views/company_branches/new.html.erb:1]: Of course!

File: app/views/employees/new.html.erb (hunk 2 above)
Comment (RobertVel666) [reply app/views/employees/new.html.erb:1]: ¡YEP!

**Comment by Arkamis** [comment]
> > Looks great, you have like 2 views with cols without row. Shouldn't you wrap all your cols inside a row tag?. And what's the difference between private number and employee number?
> 
> Yeah there are still some styles to resolve but we will be addressing them in later issues, and the private number is a number only the employee knows and is for registering the check-in/out, the employee number is just a random number assigned to an employee.

Roger but seems to me that its the same, you could use a random string and use it as password or private number and make it unique, this random string will be auto generated when a new employee is register.

Perhaps I'm wrong and you may want to let the employee decide his private number, is this the case?

**Review by Arkamis — APPROVED** [review]
Looks great :smile:

File: app/views/company_branches/new.html.erb (hunk 1 above)
Comment (anxhe) [reply app/views/company_branches/new.html.erb:1]: Of course , but this is already done in the layout:
https://github.com/BC-MAR-21-ROR/kata-07-rejoj-checador-equipo-03-k07/blob/master/app/views/layouts/application.html.erb#L15-L16 .
Doing this in the layout we can avoid repeating in every view.

**Comment by anxhe** [comment]
> > > Looks great, you have like 2 views with cols without row. Shouldn't you wrap all your cols inside a row tag?. And what's the difference between private number and employee number?
> > 
> > 
> > Yeah there are still some styles to resolve but we will be addressing them in later issues, and the private number is a number only the employee knows and is for registering the check-in/out, the employee number is just a random number assigned to an employee.
> 
> Roger but seems to me that its the same, you could use a random string and use it as password or private number and make it unique, this random string will be auto generated when a new employee is register.
> 
> Perhaps I'm wrong and you may want to let the employee decide his private number, is this the case?

I was also confused, I did not understand the need to have a random employee number, since for this case it could have been the id generated by the database, since it is a unique identifier

**Review by anxhe — APPROVED** [review]
I think that , we don't need more change in views structure styles , these are wrapped in aplication.html.erb 👍  approve

**Pull request merged**